### DISCUSSION TITLE
Is a single master cluster able to be gracefully promoted to mukti-master?

### DISCUSSION THREAD

**Discussion by 2fst4u** [k3s-io/k3s#3488] [Q&A] [answered]
The documentation describes setting up a cluster for multi-master and it sounds all well and good, but for a cluster that started life as a single master, is there a way to enable etcd and turn current worker nodes into masters?

If so, can it be done gracefully without breaking the cluster? Downtime is fine.

I have tried just running the `ks3 server --cluster-init` but it gives the following output:

```
INFO[2021-06-20T12:05:24.700498723+12:00] Starting k3s v1.21.1+k3s1 (75dba57f)
INFO[2021-06-20T12:05:24.701950330+12:00] Managed etcd cluster initializing
FATA[2021-06-20T12:05:24.762984107+12:00] starting kubernetes: preparing server: init cluster datastore and https: listen tcp :6443: bind: address already in use
```

Clearly because the server is already running. Can this be stopped, the command run and then taken over as a master again?

**Comment by brandond** [comment]
You cannot convert between datastore types (sqlite, external sql, managed etcd). You can however run a single-node etcd cluster: just start the first node with --cluster-init, and then do not add any additional servers.

**Comment by ChristianCiach** [comment, marked as answer]
Looks like this is coming: https://github.com/k3s-io/k3s/issues/3982
